### PR TITLE
[FIX](filter) update for filter_by_select logic 

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx
this pr is aim to update for filter_by_select logic and change delete limit
1. only support scala type  in delete statement where condition
2. only support column nullable and predict column support filter_by_select logic, because we can not push down non-scala type to storage layer to pack in predict column but do filter logic

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/columns/column_dictionary.h [hunk 1]
```diff
@@ -205,22 +205,6 @@ class ColumnDictionary final : public COWHelper<IColumn, ColumnDictionary<T>> {
         LOG(FATAL) << "index not implemented";
     }
 
-    Status filter_by_selector(const uint16_t* sel, size_t sel_size, IColumn* col_ptr) override {
```
Comment (yiguolei) [inline be/src/vec/columns/column_dictionary.h:208]: 这个我觉得不能删

File: be/src/vec/exec/scan/vscan_node.cpp [hunk 2]
```diff
@@ -665,6 +665,9 @@ bool VScanNode::_is_predicate_acting_on_slot(
     if (_slot_id_to_value_range.end() == entry) {
         return false;
     }
+    if (is_complex_type(remove_nullable(slot_ref->data_type()))) {
```
Comment (yiguolei) [inline be/src/vec/exec/scan/vscan_node.cpp:668]: add comment here, to explain why ??

Comment (amorynan) [reply be/src/vec/exec/scan/vscan_node.cpp:671]: done

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

File: be/src/vec/columns/column_dictionary.h (hunk 1 above)
Comment (amorynan) [reply be/src/vec/columns/column_dictionary.h]: done

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/columns/column_nullable.cpp [hunk 3]
```diff
@@ -362,8 +362,14 @@ Status ColumnNullable::filter_by_selector(const uint16_t* sel, size_t sel_size,
...
             sel, sel_size, const_cast<doris::vectorized::IColumn*>(nest_col_ptr.get())));
-    RETURN_IF_ERROR(get_null_map_column().filter_by_selector(
-            sel, sel_size, const_cast<doris::vectorized::IColumn*>(null_map_ptr.get())));
+    //insert cur nullmap into result nullmap which is empty
+    auto& res_nullmap = reinterpret_cast<vectorized::ColumnVector<UInt8>*>(
+                                const_cast<doris::vectorized::IColumn*>(null_map_ptr.get()))
+                                ->get_data();
+    DCHECK(res_nullmap.empty());
+    for (size_t i = 0; i < sel_size; i++) {
+        res_nullmap[i] = get_null_map_data()[i];
```
Comment (yiguolei) [inline be/src/vec/columns/column_nullable.cpp:371]: maybe not right??

**Comment by amorynan** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**